File: CS/SeriesDisplayNameToLegendItemVisibilityConverter.cs
using System;
using System.Windows;
using System.Windows.Data;

namespace CheckBoxesInLegendExample_UnboundMode {

    //This converter is used to hide the fake series from the legend
    class SeriesDisplayNameToLegendItemVisibilityConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            string displayName = (string)value;
            if(displayName == "fakeSeries")
                return Visibility.Collapsed;
            else
                return Visibility.Visible;
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            throw new NotImplementedException();
        }
    }

}

File: VB/SeriesDisplayNameToLegendItemVisibilityConverter.vb
Imports System
Imports System.Windows
Imports System.Windows.Data

Namespace CheckBoxesInLegendExample_UnboundMode

    'This converter is used to hide the fake series from the legend
    Friend Class SeriesDisplayNameToLegendItemVisibilityConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            Dim displayName As String = CStr(value)
            If Equals(displayName, "fakeSeries") Then
                Return Visibility.Collapsed
            Else
                Return Visibility.Visible
            End If
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace
